lib/types.ts

```diff
@@ -203,73 +203,130 @@ export type PickKeys<T, P> = Exclude<keyof PickProperties<T, P>, undefined>;
 /** Recursively omit deep properties */
 // explicitly mentioning optional properties, to work around TS making them required
 // see https://github.com/krzkaczor/ts-essentials/issues/118
-export type DeepOmit<T extends DeepOmitModify<Filter>, Filter> = T extends Builtin
+export type DeepOmit<T extends FilterModify<F>, F> = T extends Builtin
```

Comment:
I would revert it to `Filter` as then you will have nicer message if you're doing something wrong or have an issue

```suggestion
export type DeepOmit<T extends FilterModify<Filter>, Filter> = T extends Builtin
```